tests/modules/conditional_random_field_test.py

```diff
@@ -0,0 +1,36 @@
+# pylint: disable=no-self-use,invalid-name
+import torch
+from torch.autograd import Variable
+
+from allennlp.modules import ConditionalRandomField
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestConditionalRandomField(AllenNlpTestCase):
+    def test_forward_works_without_mask(self):
+        inputs = Variable(torch.Tensor([
+                [[0, 0, .5, .5, .2], [0, 0, .3, .3, .1], [0, 0, .9, 10, 1]],
+                [[0, 0, .2, .5, .2], [0, 0, 3, .3, .1], [0, 0, .9, 1, 1]],
+        ]))
+        labels = Variable(torch.LongTensor([
+                [2, 3, 4],
+                [3, 2, 2]
+        ]))
+
+        ConditionalRandomField(5, 0, 1).forward(inputs, labels)
```

Comment:
It'd be a bunch more work, but it'd also be a much better test if you set the transition weights here, and computed a known CRF loss given those transition weights, and tested against that.